port/osx/omrvmem.c

```diff
@@ -498,7 +498,7 @@ reserveMemory(struct OMRPortLibrary *portLibrary, void *address, uintptr_t byteA
 	}
 
 #if defined(AARCH64)
-	if (category->categoryCode == OMRMEM_CATEGORY_JIT_CODE_CACHE) {
+	if (0 != (mode & OMRPORT_VMEM_MEMORY_MODE_EXECUTE)) {
```

Comment:
Existing macro `OMR_ARE_ANY_BITS_SET` should be used:
https://github.com/eclipse/omr/blob/474132b797e34a3440b0e2540d2a82e5cdd8cae1/include_core/omrcomp.h#L577
```suggestion
	if (OMR_ARE_ANY_BITS_SET(mode, OMRPORT_VMEM_MEMORY_MODE_EXECUTE)) {
```

Reply:
Updated as suggested.